stix_shifter_modules/sysdig/stix_transmission/connector.py

```diff
@@ -0,0 +1,217 @@
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as ex:
+            if "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error when getting search results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'],
+                                      connector=self.connector)
+        return return_obj
+
+    @staticmethod
+    def update_event(data):
+        """
+        Set the finding_type as 'threat' for the event.
+        Replace invalid <NA> tag with empty string.
+        Parse the fd.name and set the fields for network-traffic in the event.
+        """
+        for event in data:
+            fields = event["content"]["fields"]
+            # replace unsupported field value <NA> with empty string
+            for f in fields:
+                if fields[f] == '<NA>':
+                    fields[f] = ''
+
+            # Setting the finding_type for the event as 'threat'
+            event["finding_type"] = "threat"
+            # Parsing network fields from event data
+            if fields.get("evt.type") and fields["evt.type"] == "connect":
+                if "->" in fields["fd.name"]:
+                    source, destination = fields["fd.name"].split('->')
+                    event["direction"] = "out"
+                    event["clientIpv4"], event["clientPort"] = source.split(":")
+                    event["serverIpv4"], event["serverPort"] = destination.split(":")
+                elif "<-" in fields["fd.name"]:
+                    destination, source = fields["fd.name"].split('<-')
+                    event["direction"] = "in"
+                    event["clientIpv4"], event["clientPort"] = source.split(":")
+                    event["serverIpv4"], event["serverPort"] = destination.split(":")
+                if fields.get('fd.l4proto'):
+                    event["l4protocol"] = fields['fd.l4proto']
+                else:
+                    event["l4protocol"] = "tcp"
+
+            # Adding ancestor names in list
+            fields["proc.anames"] = [val for key, val in fields.items() if 'proc.aname' in key and len(val) > 0]
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: dict
+        """
+        return_obj = {}
+        response_dict = {}
+        try:
+            response = await self.api_client.ping_data_source()
+            response_code = response.code
+            if response_code == 200:
+                return_obj['success'] = True
+            else:
+                return_obj = self.exception_response(response_code, 'cannot verify credentials')
+        except Exception as ex:
+            response_dict['message'] = ex
+            self.logger.error('error while pinging: %s', ex)
```

Comment:
Can you add connector name for all logger messages? something like `Sysdig connector error while pinging: ` or something similar or better.

Reply:
Added connector name in logger messages.